src/main/java/kr/allcll/seatfinder/pin/PinService.java

```diff
@@ -0,0 +1,29 @@
+package kr.allcll.seatfinder.pin;
+
+import java.util.List;
+import kr.allcll.seatfinder.subject.Subject;
+import kr.allcll.seatfinder.subject.SubjectRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
```

Comment:
저는 클래스 수준에 트랜잭션을 읽기 전용으로 걸고, 수정하는 메서드에서 쓰기 트랜잭션 거는 걸 선호해요. 이런 컨벤션 어떤가요? 이렇게 한다면 addPinAtSubject 메서드에 @Transactional을 명시해야 해요.

```suggestion
@Transactional(readonly = true)
```

Reply:
제가 언급한 방식을 정리해 보면, default는 읽기 전용이고 쓰기 전용을 일일이 작성해야 하는 방식이에요. 제가 생각하는 장점은 다름과 같아요! 

- 휴먼에러 방지: 쓰기 전용을 기본으로 걸어두고 읽기 전용을 메서드에 명시하게 해두면, 실수로 readonly를 명시하지 않을 경우에 알아차리기 어려워요. 반면에 읽기 전용이 기본이고 쓰기 전용을 명시하게 컨벤션을 정하면, 실수로 쓰기 전용을 명시하지 않아도 테스트 등을 통해서 휴먼에러를 방지할 수 있어요. 
- 코드량 감소: 메번 `@Transactional`를 작성하는 게, `@Transactional(readonly =true)`를 작성하는 것보다 나아요!

Reply:
아 ~~ 그럼 update를 해도 반영 안되구 그런 문제를 바로 테스트로 알 수 있으니 휴먼 에러를 방지할 수 있다는 것 맞을까요?!
조으네요
저도 그런 습관을 가져가볼게요